src/news_type.h

```diff
@@ -115,6 +115,10 @@ struct NewsTypeData {
 	NewsDisplay GetDisplay() const;
 };
 
+/** Container for any custom data that must be deleted after the news item has reached end-of-life. */
+struct NewsAllocatedData {};
+
```

Comment:
I think you have to make the dtor virtual now, as this is most likely UB :)

https://godbolt.org/z/axTW5sMEx

as example. To quote https://en.cppreference.com/w/cpp/memory/unique_ptr:

`If T is a derived class of some base B, then std::unique_ptr<T> is implicitly convertible to std::unique_ptr<B>. The default deleter of the resulting std::unique_ptr<B> will use operator delete for B, leading to undefined behavior unless the destructor of B is virtual.`

I can be completely wrong :)